### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by AbrahamGeorge8547** [dynamoose/dynamoose#1209]
### Summary:
Not able to use update with $ADD with typescript

### Code sample:
`await clientModel.update({'id': clientId}, {"$ADD": {"facilities": [facilityId]}})`
#### Schema
```
`const clientSchema = new Schema(
  {
    id: { hashKey: true, type: String, default: v4() },
    name: String,
    phone: { type: String },
    email: { type: String },
    // ownedQueues: [{ type: Types.ObjectId, ref: 'QueueTemplate' }],
    // controlledQueues: [{ type: Types.ObjectId, ref: 'QueueTemplate' }],
    facilities: [String]
  },
  { timestamps: true }
);`
// Code here
```
Interface:
`export interface IClient extends Document {
  id: string;
  name: string;
  phone: string;
  email: string;
  ownedQueues?: string[];
  controlledQueues?: string[];
  facilities?: string[];
}`

### Current output and behavior (including stack trace):
`No overload matches this call.
  Overload 1 of 10, '(obj: Partial<IClient>, callback: CallbackType<IClient, AWSError>): void', gave the following error.
    Argument of type '{ $ADD: { facilities: string[]; }; }' is not assignable to parameter of type 'CallbackType<IClient, AWSError>'.
      Object literal may only specify known properties, and '"$ADD"' does not exist in type 'CallbackType<IClient, AWSError>'.
  Overload 2 of 10, '(keyObj: ObjectType, updateObj: Partial<IClient>): Promise<IClient>', gave the following error.
    Argument of type '{ $ADD: { facilities: string[]; }; }' is not assignable to parameter of type 'Partial<IClient>'.
      Object literal may only specify known properties, and '"$ADD"' does not exist in type 'Partial<IClient>'`

### Expected output and behavior:

update need to happen

### Environment:

Operating System:
Operating System Version:
Node.js version (`node -v`): v12.13.0
NPM version: (`npm -v`):6.12.0
Dynamoose version:2.7.3

### Other:
- 

- [x] I have read through the Dynamoose documentation before posting this issue

- [x] I have searched through the GitHub issues (including closed issues) and pull requests to ensure this issue has not already been raised before
- [x] I have searched the internet and Stack Overflow to ensure this issue hasn't been raised or answered before
- [x] I have tested the code provided and am confident it doesn't work as intended
- [x] I have filled out all fields above
- [x] I am running the latest version of Dynamoose

**Comment by RationalG** [comment]
This issue makes Dynamoose unusable if atomic counters are needed. +1

**Comment by lmanerich** [comment]
A workaround is to create transaction for this update:

```ts
await dynamoose.transaction(clientModel.transaction.update({'id': clientId}, {"$ADD": {"facilities": [facilityId]}}));
```

**Comment by kevinlbatchelor** [comment]
The documentation shows that you can do atomic updates.

From the docs

`await User.update({"id": 1}, {"$SET": {"name": "Bob"}, "$ADD": {"age": 1}});`

Bur reading through the code I don't see anything to handle this. I think this is a bad docs?

**Comment by fishcharlie** [comment]
> Bur reading through the code I don't see anything to handle this. I think this is a bad docs?

Nope. DynamoDB handles this for us. No need to do anything on our end.

**Comment by fishcharlie** [comment]
I can't seem to reproduce this issue. I just added another type test to the library. Maybe a recent version of Dynamoose has fixed this. If anyone is able to create a reproducible example I'd be happy to look at this further. Otherwise this issue will be closed soon.

See my added test in this commit: c4f5fbf25bc47f5b88b1dbf946a6a4198323b5f8
